agentendpoint/config_task.go

```diff
@@ -402,14 +295,41 @@ func (c *configTask) run(ctx context.Context) error {
 	// We need to generate base results first thing, each execution step
 	// just adds on.
 	c.generateBaseResults()
-	c.validation(ctx)
 	defer c.cleanup(ctx)
-	c.checkState(ctx)
 
 	if err := c.reportContinuingState(ctx, agentendpointpb.ApplyConfigTaskProgress_APPLYING_CONFIG); err != nil {
 		return c.handleErrorState(ctx, rcsErrMsg, err)
 	}
-	c.enforceState(ctx)
+
+	c.policies = map[string]*policy{}
+	for i, osPolicy := range c.Task.GetOsPolicies() {
+		ctx = clog.WithLabels(ctx, map[string]string{"config_assignment": osPolicy.GetOsPolicyAssignment(), "policy_id": osPolicy.GetId()})
+		clog.Debugf(ctx, "Executing policy:\n%s", util.PrettyFmt(osPolicy))
+
+		pResult := c.results[i]
+		plcy := &policy{resources: map[string]resourceIface{}}
+		c.policies[osPolicy.GetId()] = plcy
+		var policyMR *config.ManagedResources
+
+		for i, configResource := range osPolicy.GetResources() {
+			rCompliance := pResult.GetOsPolicyResourceCompliances()[i]
+			validateConfigResource(ctx, plcy, policyMR, rCompliance, configResource)
+			if plcy.hasError {
+				break
+			}
+			checkConfigResourceState(ctx, plcy, rCompliance, configResource)
+			if plcy.hasError {
+				break
+			}
+			if enforceConfigResourceState(ctx, plcy, rCompliance, configResource) {
```

Comment:
Consider naming the return value for this method as it could be potentially confusing while reading to determine if it is indicating whether enforcement was attempted or whether enforcement was successful

Reply:
Done here and in the function